src/xcode/ENA/ENA/Source/AppDelegate & Globals/NotificationManager.swift

```diff
@@ -128,65 +126,68 @@ final class NotificationManager: NSObject, UNUserNotificationCenterDelegate {
-			let route = Route(healthCertifiedPerson: certifiedPerson)
-			Log.debug("Received boosterVaccination notification")
-			showHealthCertifiedPerson(route)
-		} else if let (certifiedPerson) = extractPerson(LocalNotificationIdentifier.certificateReissuance.rawValue, from: identifier) {
-			let route = Route(healthCertifiedPerson: certifiedPerson)
-			Log.debug("Received certificateReissuance notification")
-			showHealthCertifiedPerson(route)
-		} else if let (certifiedPerson) = extractPerson(LocalNotificationIdentifier.admissionStateChange.rawValue, from: identifier) {
-			let route = Route(healthCertifiedPerson: certifiedPerson)
-			Log.debug("Received admissionStateChange notification")
-			showHealthCertifiedPerson(route)
-		} else if let (certifiedPerson, healthCertificate) = extract(LocalNotificationIdentifier.certificateBlocked.rawValue, from: identifier) {
-			let route = Route(
-				healthCertifiedPerson: certifiedPerson,
-				healthCertificate: healthCertificate
-			)
-			Log.debug("Received Certificate blocked notification")
-			showHealthCertificate(route)
+		
+	private func checkForLocalNotificationsActions(_ incomingIdentifier: String) {
+		guard let certificateIdentifier = LocalNotificationIdentifier.allCases.first(where: {
+			incomingIdentifier.hasPrefix($0.rawValue)
+		}) else {
+			return
+		}
+		
+		switch certificateIdentifier {
+		case .certificateExpired, .certificateExpiringSoon, .certificateBlocked, .certificateInvalid:
+			extract(certificateIdentifier.rawValue, from: incomingIdentifier, completion: { [weak self] result in
+				if let (certifiedPerson, healthCertificate) = result {
+					let route = Route(
+						healthCertifiedPerson: certifiedPerson,
+						healthCertificate: healthCertificate
+					)
+					Log.debug("Received \(certificateIdentifier.rawValue) notification")
+					self?.showHealthCertificate(route)
+				}
+			})
+		case .boosterVaccination, .certificateReissuance, .admissionStateChange:
+			extractPerson(certificateIdentifier.rawValue, from: incomingIdentifier, completion: { [weak self] result in
+				if let certifiedPerson = result {
+					let route = Route(healthCertifiedPerson: certifiedPerson)
+					Log.debug("Received \(certificateIdentifier.rawValue) notification")
+					self?.showHealthCertifiedPerson(route)
+				}
+			})
+		case .checkout:
+			break
 		}
 	}
 	
-	private func findHealthCertificate(_ identifier: String) -> (HealthCertifiedPerson, HealthCertificate)? {
-		for person in healthCertificateService.healthCertifiedPersons {
-			if let certificate = person.$healthCertificates.value
-				.first(where: { $0.uniqueCertificateIdentifier == identifier }) {
-				return (person, certificate)
+	private func findHealthCertificate(_ identifier: String, completion: @escaping((HealthCertifiedPerson, HealthCertificate)?) -> Void) {
+		healthCertificateService.setup(updatingWalletInfos: true) { [weak self] in
```

Comment:
Can it happen, that we now call healthCertificateService.setup twice? Because setup is also called on this instance from the app delegate.

Reply:
@ArturFriesen  I just checked if the app is in either foreground, background or terminated mode and we dont call the healthCertificateService.setup() in appdelegate and all the 3 cases so it appears no.
I also added a dispatch statement to show the UI from the main thread